include/deal.II/multigrid/mg_constrained_dofs.h

```diff
@@ -100,6 +100,21 @@ class MGConstrainedDoFs : public Subscriptor
     const std::set<types::boundary_id> &boundary_ids,
     const ComponentMask &               component_mask = ComponentMask());
 
+  /**
+   * Fill the internal data structures with information
+   * about no normal flux boundary dofs.
+   *
+   * This function is limited to meshes whose no normal flux boundaries
+   * have faces which are normal to the x, y, or z axis. Also, when generating
+   * the mesh, the user must have set colorize=true and call
+   * make_no_normal_flux_constraints() for each boundary id associated with no
+   * normal flux conditions separately."
+   */
+  template <int dim, int spacedim>
+  void
+  make_no_normal_flux_constraints(const DoFHandler<dim, spacedim> &dof,
```

Comment:
I just realized that this interface assumes that the vector is considered component 0, .., dim, which is of course not the general case. Can we add ``	const unsigned int 	first_vector_component``?